pkg/pulumiyaml/run.go

```diff
@@ -773,7 +825,47 @@ func (e programEvaluator) EvalOutput(r *Runner, node ast.PropertyMapEntry) bool
 
 func (r *Runner) Evaluate(ctx *pulumi.Context) syntax.Diagnostics {
 	eCtx := r.newContext(nil)
-	return r.Run(programEvaluator{evalContext: eCtx, pulumiCtx: ctx})
+
+	// Register package refs for all packages we know upfront
+	packageRefs := make(map[tokens.Package]string)
+	for _, pkg := range r.packageDescriptors {
+		name := pkg.Name
+		// If parametrized use the parametrized name
+		if pkg.Parameterization != nil {
+			name = pkg.Parameterization.Name
+		}
+
+		var parameterization *pulumirpc.Parameterization
+		if pkg.Parameterization != nil {
```

Comment:
Minor but we could combine these ifs to avoid the double check (cheap as it is). Happy either way. Same is true of `ToPackageDescriptors` earlier too if we feel like it.

Reply:
Sure